code/campaignclassic/src/main/java/com/adobe/marketing/mobile/AEPPushTemplateBroadcastReceiver.java

```diff
@@ -0,0 +1,302 @@
+        leftButtonIntent.putExtra(CampaignPushConstants.IntentKeys.RIGHT_IMAGE, newRightImageUri);
+        leftButtonIntent.putExtra(CampaignPushConstants.IntentKeys.LEFT_CAPTION, newLeftCaption);
+        leftButtonIntent.putExtra(
+                CampaignPushConstants.IntentKeys.CENTER_CAPTION, newCenterCaption);
+        leftButtonIntent.putExtra(CampaignPushConstants.IntentKeys.RIGHT_CAPTION, newRightCaption);
+        leftButtonIntent.putExtra(CampaignPushConstants.IntentKeys.SMALL_TITLE_TEXT, titleText);
+        leftButtonIntent.putExtra(CampaignPushConstants.IntentKeys.SMALL_BODY_TEXT, smallBodyText);
+        leftButtonIntent.putExtra(
+                CampaignPushConstants.IntentKeys.EXPANDED_BODY_TEXT, expandedBodyText);
+        leftButtonIntent.putExtra(CampaignPushConstants.IntentKeys.CHANNEL_ID, channelId);
+        leftButtonIntent.putExtra(CampaignPushConstants.IntentKeys.MESSAGE_ID, messageId);
+
+        final Intent rightButtonIntent =
+                new Intent(
+                        CampaignPushConstants.IntentActions.FILMSTRIP_RIGHT_CLICKED,
+                        null,
+                        context,
+                        AEPPushTemplateBroadcastReceiver.class);
+        rightButtonIntent.setClass(context, AEPPushTemplateBroadcastReceiver.class);
+        rightButtonIntent.setFlags(Intent.FLAG_ACTIVITY_SINGLE_TOP);
+        rightButtonIntent.putExtra(
+                CampaignPushConstants.IntentKeys.BACKGROUND_COLOR, backgroundColorHex);
+        rightButtonIntent.putExtra(CampaignPushConstants.IntentKeys.TITLE_COLOR, titleColorHex);
+        rightButtonIntent.putExtra(CampaignPushConstants.IntentKeys.BODY_COLOR, bodyColorHex);
+        rightButtonIntent.putExtra(CampaignPushConstants.IntentKeys.ICON_COLOR, iconColorHex);
+        rightButtonIntent.putExtra(CampaignPushConstants.IntentKeys.LEFT_IMAGE, newLeftImageUri);
+        rightButtonIntent.putExtra(
+                CampaignPushConstants.IntentKeys.CENTER_IMAGE, newCenterImageUri);
+        rightButtonIntent.putExtra(CampaignPushConstants.IntentKeys.RIGHT_IMAGE, newRightImageUri);
+        rightButtonIntent.putExtra(CampaignPushConstants.IntentKeys.LEFT_CAPTION, newLeftCaption);
+        rightButtonIntent.putExtra(
+                CampaignPushConstants.IntentKeys.CENTER_CAPTION, newCenterCaption);
+        rightButtonIntent.putExtra(CampaignPushConstants.IntentKeys.RIGHT_CAPTION, newRightCaption);
+        rightButtonIntent.putExtra(CampaignPushConstants.IntentKeys.SMALL_TITLE_TEXT, titleText);
+        rightButtonIntent.putExtra(CampaignPushConstants.IntentKeys.SMALL_BODY_TEXT, smallBodyText);
+        rightButtonIntent.putExtra(
+                CampaignPushConstants.IntentKeys.EXPANDED_BODY_TEXT, expandedBodyText);
+        rightButtonIntent.putExtra(CampaignPushConstants.IntentKeys.CHANNEL_ID, channelId);
+        rightButtonIntent.putExtra(CampaignPushConstants.IntentKeys.MESSAGE_ID, messageId);
```

Comment:
If we use the array and center index approach, you can reuse the same intent for both the left and right button click intent because the notification builder is able to use the direction of the click and the current center image to figure out the next notification to build.